rainmaker_orchestrator/server.py

```
@@ -1,426 +1,628 @@
-            # (the main loop may already be closed)
-            loop = asyncio.new_event_loop()
-            asyncio.set_event_loop(loop)
-            try:
-                loop.run_until_complete(orchestrator.aclose())
-                logger.info("Orchestrator HTTP client closed successfully")
-            finally:
-                loop.close()
+        dir_path = Path(settings.workspace_path) / path
+
+        if not dir_path.is_relative_to(settings.workspace_path):
+            raise HTTPException(status_code=403, detail="Access denied")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)